_posts/2018-05-18-contributing-to-installer-puppet-modules.md

```diff
@@ -0,0 +1,326 @@
+
+```
+PLAY RECAP *********************************************************************
+centos7-katello-nightly    : ok=19   changed=12   unreachable=0    failed=0   
+
+Wednesday 16 May 2018  11:18:53 -0400 (0:00:00.012)       0:16:52.838 ********* 
+=============================================================================== 
+foreman_installer : Run installer ------------------------------------- 537.06s
+foreman_installer : Install additional packages ----------------------- 344.72s
+foreman_installer : Install foreman-installer -------------------------- 50.66s
+foreman_repositories : Install foreman-release-scl --------------------- 34.82s
+selinux : Ensure libselinux-python is installed ------------------------ 26.23s
+epel_repositories : Setup Epel Repository ------------------------------- 6.79s
+foreman_repositories : Setup Foreman Repository ------------------------- 6.34s
+katello_repositories : Setup Katello Repository ------------------------- 2.23s
+puppet_repositories : Setup Puppet 5 Repository ------------------------- 1.62s
+Gathering Facts --------------------------------------------------------- 0.71s
+selinux : Set selinux state --------------------------------------------- 0.45s
+etc_hosts : Build hosts file -------------------------------------------- 0.33s
+epel_repositories : Enable CentOS-CR Repository ------------------------- 0.30s
+ostree_repositories : enable-ostree-repo -------------------------------- 0.30s
+katello_repositories : Set up release repositories ---------------------- 0.03s
+foreman_repositories : include_tasks ------------------------------------ 0.03s
+puppet_repositories : Include puppet 5 for RedHat ----------------------- 0.03s
+foreman_installer : include_tasks --------------------------------------- 0.03s
+foreman_installer : include_tasks --------------------------------------- 0.03s
+katello_repositories : include_tasks ------------------------------------ 0.02s
+```
+
+If the command was not successful, please reach out to the [foreman-dev discourse list](https://community.theforeman.org/c/development) (preferred), or #theforeman-dev on freenode. We love having new contributors; someone will help you. It's possible that you did everything correctly and there is a bug in forklift, so don't be afraid to reach out for assistance.
+
+Once you have your centos7-katello-nightly VM running, you are ready to develop!  The next step is to ssh to your instance with `vagrant ssh centos7-katello-nightly`. You should be presented with a shell, and can attempt to reproduce the issue. In our example, we will want to add an option to optionally remove indexes from the `/pub` directory.
+
+The first step is always to reproduce the issue on the nightly VM. If our hypervisor is not our workstation/laptop, we will need to get https and http access to the nightly VM first. To do this we'll set up an SSH tunnel on our hypervisor. 
+
+First, capture the IP address of the guest:
```

Comment:
This may or may not be valid here, but it can be useful info to have as a way to get similar info:

```
ehelms@woodford forklift (fixes-677)$ vagrant ssh-config centos7-luna-devel
Host centos7-luna-devel
  HostName 192.168.121.229
  User vagrant
  Port 22
  UserKnownHostsFile /dev/null
  StrictHostKeyChecking no
  PasswordAuthentication no
  IdentityFile /home/ehelms/workspace/forklift/.vagrant/machines/centos7-luna-devel/libvirt/private_key
  IdentitiesOnly yes
  LogLevel FATAL
```